src/main/java/org/kuali/kra/budget/document/BudgetDocumentRule.java

```diff
@@ -154,7 +154,86 @@ protected boolean processCustomSaveDocumentBusinessRules(Document document) {
     * @param budgetDocument
     * @return
     */
-    @SuppressWarnings("rawtypes")
+    @SuppressWarnings("rawtypes")  
+	protected boolean processBudgetProjectIncomeBusinessRule(BudgetDocument budgetDocument) {
+        boolean valid = true;
+        MessageMap errorMap = GlobalVariables.getMessageMap();
+        int i = 0;
+        
+        // Get each budget cost share.
+        for (BudgetCostShare budgetCostShare : budgetDocument.getBudget().getBudgetCostShares()) {
+            String errorPath = "budgetCostShare[" + i + "]";
+            errorMap.addToErrorPath(errorPath);
+            
+            // Validate share percentage.
+            if(budgetCostShare.getSharePercentage() != null && (budgetCostShare.getSharePercentage().isLessThan(new BudgetDecimal(0)) || 
+               budgetCostShare.getSharePercentage().isGreaterThan(new BudgetDecimal(100)))) {
```

Comment:
Two magic numbers, consider placing into class constants with a variable name describing what its intent is.